File: jiWoo/프로그래머스_전화번호목록.java
package jiWoo;
import java.util.*;
public class 프로그래머스_전화번호목록 {
    public boolean solution(String[] phone_book) {
        boolean answer = false;
        Arrays.sort(phone_book); // 숫자 문자열을 정렬하면 사전순으로 정렬된다.
        for(int i=0; i<phone_book.length-1; i++){ // 그러므로 뒤 문자만 비교
            int len=phone_book[i].length();
            // 다음 전화번호가 현재 번호보다 길고 잘랐을 때 같으면 false
            if(phone_book[i+1].length()>=len && phone_book[i+1].substring(0,len).equals(phone_book[i])){
                return answer;
            }
        }
        answer=true;
        return answer;
    }
}

File: yunhwan/프로그래머스 전화번호 목록.py
def solution(phone_book):
    answer = True
    
    #정렬
    phone_book.sort()
    
    #순회
    for i in range(len(phone_book)-1):
        #현재 전화번호의 길이만큼, 다음문자열 자르기
        #자른 다음 전화번호에 있는지 확인
        if phone_book[i] in phone_book[i+1][:len(phone_book[i])]:
            return False
    
    
    return answer